libs/core/src/CutFlowManager.cpp

```diff
@@ -40,15 +40,19 @@ void CutFlowManager::RegisterPreExistingCutFlows() {
     if (!eventReader->inputFile->Get(cutFlowName.c_str())) continue;
     info() << "Input file contains " << cutFlowName << " directory - will store existing cutflow in the output." << endl;
 
+    bool rawEvents = cutFlowName.find("RawEvents")!=string::npos;
     string collectionName = "";
-    if(cutFlowName=="CutFlow") collectionName = "";
-    else if(cutFlowName.find("CollectionCutFlow_")!=string::npos) collectionName = cutFlowName.substr(cutFlowName.find("CollectionCutFlow_")+17);
+    if(cutFlowName=="CutFlow" || cutFlowName=="RawEventsCutFlow") collectionName = "";
+    else if(cutFlowName.find("CollectionCutFlow_")!=string::npos) {
+      collectionName = cutFlowName.substr(cutFlowName.find("CollectionCutFlow_")+17);
+      if(rawEvents) collectionName = collectionName.substr(collectionName.find("RawEventsCollectionCutFlow_")+26);
+    }
     else {
       warn() << "Found cutflow in unknown format: " << cutFlowName << ". Expected cutflow name CutFlow or names starting with CollectionCutFlow_" << endl;
       collectionName = cutFlowName;
     }
 
-    RegisterCollection(collectionName);
+    if(!rawEvents && collectionName!="") RegisterCollection(collectionName);
```

Comment:
Maybe I missed something, but shouldn't we also register collections with raw events?

Reply:
I missed to include the rawEventsAfterCollectionCuts in the RegisterCollection. But I added it now so once you register the collection for weights everything you need for rawEvents is also registered, I just set it up like this  so you don't register some collection variables twice.